CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -840,6 +840,31 @@ static function BarracksStatusReport GetBarracksStatusReport()
 	return CurrentBarracksStatus;
 }
 
+static function string GetPostMissionText (XComGameState_Activity ActivityState, bool bVictory)
+{
+	local XComGameState_ActivityChain ChainState;
+	local string ChainName;
+
+	ChainState = ActivityState.GetActivityChain();
+	ChainName = ChainState.GetMyTemplate().strTitle;
+
+	if (bVictory)
+	{
+		if (ChainState.GetLastActivity().GetMyTemplate() == ActivityState.GetMyTemplate())
+		{
+			return "'" $ ChainName $ "' Chain Completed";
+		}
+		else
+		{
+			return "'" $ ChainName $ "' Chain Progressed";
+		}
+	}
+	else
+	{
+		return "'" $ ChainName $ "' Chain Failed";
```

Comment:
That isn't a necessity though.... It all depends on what `Template.ShouldProgressChain()` returns.

In fact, the chain framework itself has no concept of a "failed chain" - only one with a `eACER_ProgressBlocked` - although that's how it's currently used....

IMO we should add some vars to the activity template or just ignore this "issue" altogether and just leave this version (with assumptions).

In any case, **localization**

Reply:
If the chain is not supposed to end on a failed activity, then the activity template can specify a special OnFailure function that displays a different message.

Reply:
My dreams of chain-agnostic activities shattered -.-